build.sbt

```diff
@@ -42,12 +42,28 @@ lazy val core = crossProject(JVMPlatform, JSPlatform)
   .settings(
     name := "cats-effect-cps",
 
-    scalacOptions += "-Xasync",
+    scalacOptions ++= {
+      if (isDotty.value)
+        Seq()
+      else
+        Seq("-Xasync")
+    },
+
+    Compile / unmanagedSourceDirectories +=
+      (Compile / baseDirectory).value.getParentFile() / "shared" / "src" / "main" / s"scala-${scalaVersion.value.split('.').head}",
+
+    Test / unmanagedSourceDirectories +=
+      (Test / baseDirectory).value.getParentFile() / "shared" / "src" / "main" / s"scala-${scalaVersion.value.split('.').head}",
 
     libraryDependencies ++= Seq(
-      "org.typelevel" %% "cats-effect-std" % CatsEffectVersion,
-      "org.scala-lang" % "scala-reflect"   % scalaVersion.value % "provided",
+      "org.typelevel" %%% "cats-effect-std" % CatsEffectVersion,
+
+      "org.typelevel" %%% "cats-effect"                % CatsEffectVersion % Test,
+      "org.typelevel" %%% "cats-effect-testing-specs2" % "1.1.1"           % Test),
 
-      "org.typelevel" %% "cats-effect"                % CatsEffectVersion % Test,
-      "org.typelevel" %% "cats-effect-testing-specs2" % "1.1.1"           % Test,
-      "org.specs2"    %% "specs2-core"                % "4.12.1"          % Test))
+    libraryDependencies ++= {
+      if (isDotty.value)
+        Seq("com.github.rssh" %%% "dotty-cps-async" % "0.8.1")
```

Comment:
Here are my thoughts (re-iterated here for the sake of visibility) : `dotty-cps-async` is a super cool library, but before we commit to depending on it, I'd like to know more about the status/plan for it. Is it sponsored commercially ? Is it a research project conducted by a small bunch of individuals for the sake of proving it's possible ? Are there (even loose) plans for it to be integrated into the Scala 3 standard library ? (or even moved over to the `scala` organisation). (@rssh could you provide insight ?)

My concern is that the AST transformation this library performs is not trivial by any mean. If it was to become unmaintained (for any reason whatsoever), we'd be in the awkward position of having to assume maintenance burden, burden which is increased by the currently poor state of tooling and documentation around metaprogramming in Scala 3. 

Of course one could say "we could always give up on `cats-effect-cps` if that happens", but if people actually start to use it, I for one will experience the self-inflicted pressure to assume the maintenance cost. 

I'm talking from experience here, as I'm now participating (with @keynmol) to the maintenance of https://github.com/eed3si9n/expecty, another macro-heavy library, because weaver depends on it and Eugene doesn't have the capacity to maintain it properly.

Reply:
1 - yes. Note, that the company behind is mine.
2 - not clear, what is 'research project',  how you differ research project from non-research?
3 - out of my control.  From my estimations of the scala-center processes - no, but better ask them directly ;)
4. - out of my control.

Reply:
I definitely agree that this is not a dependency we would take on lightly. I think it's okay to pull it into cats-effect-cps for now, particularly since we have giant "EXPERIMENTAL" flags all over everything, but before we pull this into CE core we're going to need to have a larger conversation.

My *guess* is that this is indeed something that both Scala Center and EPFL would be interested in. I'm making that guess for a couple of reasons:

- This dovetails very closely with some @odersky's current research into effect systems, and indeed provides a pathway for that research to be generalized to existing systems like Cats Effect
- Certain major companies which rely on Scala 2 will need this kind of functionality to even *consider* Scala 3
- There's a lot of community interest in this type of syntactic construct, which is an obvious motivator for not only SC but also other major partners such as VirtusLabs and 47 Degrees

Tldr, I would be very surprised if something like this *didn't* end up in the Scala 3 compiler at some point. *When* and in *what form* are more open questions. It would also be lovely to hear from more authoritative voices than mine on some of the future thinking around this.

Reply:
>  but before we pull this into CE core we're going to need to have a larger conversation. 

Alright, gave a fair bit of thoughts to this. Given that the merge into CE would be conditioned by not-only the stabilisation of `-Xasync` on Scala2, but also by the gathering of enough evidence that this syntax is actually used, I think I'm okay to accept the dependency at this point in time, as it will give us more data points.